functions_lab.js

```diff
@@ -0,0 +1,283 @@
+// # Function Exercises
+//
+// ## Question 1
+//
+// Write a function that returns the largest of two numbers. If they are equal, return the second one.
+//
+// Example:
+//
+// Input: `12, 4`
+//
+// Output: `12`
+console.log('----- Question 1 -----');
+const compare = (num1, num2) => {
+  if (num1 > num2) {
+    return num1;
+  } else {
+    return num2;
+  }
+}
+
+console.log(compare(300, 300));
+
+// ## Question 2
+//
+// Write a function named getDogAge that takes one argument: a dog's age in human years.
+//The function returns the dog's age in dog years. The conversion rate is 1 human year to 7 dog years.
+//
+// Example:
+//
+// Input: `4`
+//
+// Output: `28`
+console.log('----- Question 2 -----');
+const getDogAge = (age) => {
+  return age * 7;
+}
+
+console.log(getDogAge(9));
+
+// ## Question 3
+//
+// Write a function `arraySum` that takes in an array and returns the sum of all the elements.
+//
+// Example:
+//
+// Input: `[13,2,4,-2]`
+//
+// Output: `17`
+console.log('----- Question 3 -----');
+let number = [1, 2, 3, 4, 5];
+const arraySum = (array) => {
+  let sum = 0;
+  for (i = 0; i < number.length; i++) {
+    sum += number[i];
+  }
+
+  return sum;
+}
+let total = arraySum(number);
+console.log(total);
+
+// ## Question 4
+//
+// Write a function that takes in any two numbers and return their sum.
+//
+// Example:
+//
+// Input: `14, 5`
+//
+// Output: `19`
+console.log('----- Question 4 -----');
+
+let negOneHundred = -100;
+let eightyFive = 85;
+const sum = (num1, num2) => {
+  return num1 + num2;
+}
+
+console.log(sum(negOneHundred, eightyFive));
+
+// ## Question 5
+//
+// Write a function takes in any number grade and returns a corresponding letter grade.
+//
+// | Number Grade | Equivalent Letter Grade |
+// | :--------: | :---------: |
+// | 100 | A+ |
+// | 90 - 99 | A |
+// | 80 - 89 | B |
+// | 70 - 79 | C |
+// | 65 - 69 | D |
+// | Below 65 | F |
+//
+//
+// Example:
+//
+// Input: `88`
+//
+// Output: `"B"`
+console.log('----- Question 5 -----');
+const gradeConversion = (grade) => {
+  if (grade === 100) {
+    return "'A+'";
+  } else if (grade > 89) {
+    return "'A'";
+  } else if (grade > 79) {
+    return "'B'";
+  } else if (grade > 69) {
+    return "'C'";
+  } else if (grade >= 65) {
+    return '"D"';
+  } else {
+    return '"F"';
+  }
+}
+
+console.log(gradeConversion(65));
+
+// ## Question 6
+//
+// Implement a function named `repeatLog` that takes a string and a number as parameters. The function should log `message` `count` number of times.
+//
+// Example:
+//
+// Input: `"+", 10`
+//
+// Output: `++++++++++`
+//
+console.log('----- Question 6 -----');
+let str = '';
+const repeatLog = (string, num) => {
```

Comment:
You could declare str variable inside the function and return str instead. It works but it's better to minimize side effects when possible.